src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -445,6 +445,8 @@ void remove_virtual_functionst::get_functions(
         has_prefix(
           id2string(b.symbol_expr.get_identifier()), "java::java.lang.Object"))
         return true;
+      else if(a.symbol_expr.get_identifier() == b.symbol_expr.get_identifier())
```

Comment:
So if the function names are equal then sort by class names? Don't the function names include the class names though?

Reply:
`class_id` is the name of the class, whereas `symbol_expr` is the name of the target method to call (which might be defined in a parent class). We are sorting by target method and if the same target method is used by several classes (these generate the big or expression in the dispatch condition)  then we sort by these classes.